content/docs/extending-tina/custom-field-components.md

```diff
@@ -0,0 +1,114 @@
+---
+title: 'Custom field components'
+id: '/docs/extending-tina/custom-field-components'
+prev: /docs/extending-tina/validation
+next: /docs/extending-tina/customize-list-ui
+---
+
+A custom component can be passed and rendered by providing `ui.component` property in a field. This component completely overrides the original component, providing the user with the full ability to customize the field.
+
+## Providing your own component
+
+A fields `ui.component` property is a React component that accepts three props:
+
+- `field`: The [field definition](https://tinacms.org/docs/reference/toolkit/fields) for the current field.
+- `input`: The data and callbacks necessary to make an input.
+- `meta`: Metadata about the field in the form. (e.g. `dirty`, `valid`)
+
+Checkout the [react-final-form](https://github.com/final-form/react-final-form#fieldrenderprops) docs for a more detailed description of the `input` and `meta` props.
+
+
+Let's create a basic slider component:
```

Comment:
does this section need a sub-header (maybe "Example" or "Custom Component Example")?

Reply:
It also might be good to bring the image of the component up above the code and provide a little blurb about what the component is intended for. Something like: "Here is a custom slider component that can be used for adjusting image saturation"

Reply:
Yup totally!